src/main/java/traincraft/blocks/assemblytables/SlotCraftingResult.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ *
+ * This file ("SlotCraftingResult.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ */
+package traincraft.blocks.assemblytables;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryCraftResult;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.CraftingManager;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.NonNullList;
+import org.apache.logging.log4j.Level;
+import traincraft.Traincraft;
+
+/**
+ * This is similar to Mojang's SlotCrafting, except it is tailored specifically to be used with the assemblyTable.
+ * NOTE: This will only work correctly if it's used with AssemblyTableRecipes.
+ *
+ * @author PseudonymPatel
+ * @since 2020-8-3
+ */
+public class SlotCraftingResult extends Slot {
+    
+    private final AssemblyCraftingItemHandler craftMatrix;
+    private final EntityPlayer player;
+    private int amountCrafted;
+    
+    public SlotCraftingResult(EntityPlayer player, AssemblyCraftingItemHandler craftingInventory, IInventory inventoryIn, int slotIndex, int xPosition, int yPosition){
+        super(inventoryIn, slotIndex, xPosition, yPosition);
+        this.player = player;
+        this.craftMatrix = craftingInventory;
+    }
+    
+    @Override
+    protected void onCrafting(ItemStack stack, int amount){
+        this.amountCrafted += amount;
+        this.onCrafting(stack);
+    }
+    
+    @Override
+    protected void onSwapCraft(int p_190900_1_){
+        this.amountCrafted += p_190900_1_;
+    }
+    
+    @Override
+    protected void onCrafting(ItemStack stack){
+        if(this.amountCrafted > 0){
+            stack.onCrafting(this.player.world, this.player, this.amountCrafted);
+            //TODO: see if neccesary
+            
+            //net.minecraftforge.fml.common.FMLCommonHandler.instance().firePlayerCraftingEvent(this.player, stack, craftMatrix);
```

Comment:
shouldn't be needed, since this isn't a crafting table

Reply:
ok, thanks